### PR TITLE
Add floating header for progress table

### PR DESCRIPTION
**Pull request by lfryemason**
Adds a floating header to the top of the progress table.

This floating header is a copy of the header of the progress table that only shows up when the header that is at the top of the table is not visible. This copy is still controlled by the floating horizontal scrollbar for the table.

https://github.com/code-dot-org/code-dot-org/assets/25193259/d51dc313-18f5-4f79-8dec-7d7e234876b2

## Links
[TEACH-940](https://codedotorg.atlassian.net/browse/TEACH-940)

## Testing story

* Testing manually both locally and on [adhoc](https://adhoc-lfm-p-flt-studio.cdn-code.org/) 
* Unit tests for the header copy

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgressV2/LockedLessonUtils.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    sortedStudents.map(student => [
+      student.id,
+      lesson.lockable &&
+        lesson.levels.every(
+          level => levelProgressByStudent[student.id][level.id]?.locked
+        ),
+    ])
+  );
+
+export const areAllLevelsLocked = lockedStatusPerStudent => {
```
Comment (kobryan0619) [inline apps/src/templates/sectionProgressV2/LockedLessonUtils.jsx:16]: Out of curiosity, why is this needed when we have LockedStatusPerStudent?

File: apps/src/templates/sectionProgressV2/floatingHeader/FloatingHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import LessonProgressColumnHeader from '../LessonProgressColumnHeader';
+import {
+  areAllLevelsLocked,
+  getLockedStatusPerStudent,
+} from '../LockedLessonUtils';
+
+import styles from './floating-header.module.scss';
+import progressStyles from '../progress-table-v2.module.scss';
+
+/** A component that displays lesson headers in a floating header above a table.
```
Comment (kobryan0619) [inline apps/src/templates/sectionProgressV2/floatingHeader/FloatingHeader.jsx:16]: I appreciate these comments for context

File: apps/src/templates/sectionProgressV2/floatingHeader/FloatingHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            />
+          </div>
+        );
+      }),
+    [lessons, expandedLessonIds, getLessonAllLocked]
+  );
+
+  return (
+    // The floating header is a copy of the header that isn't visible.
+    // Screen readers should ignore it and only need the original/non-floating header..
```
Comment (kobryan0619) [inline apps/src/templates/sectionProgressV2/floatingHeader/FloatingHeader.jsx:161]: Nice accessibility comment here.

File: apps/src/templates/sectionProgressV2/LockedLessonUtils.jsx (hunk 1 above)
Comment (lfryemason) [reply apps/src/templates/sectionProgressV2/LockedLessonUtils.jsx:16]: This is determining whether all levels on a lesson are locked for each student instead of for one student

**Review by kobryan0619 — APPROVED** [review]
Looks good - really meaty content here!
